src/estimagic/optimization/tranquilo/region.py

```diff
@@ -0,0 +1,96 @@
+from dataclasses import dataclass, replace
+
+import numpy as np
+
+from estimagic.optimization.tranquilo.bounds import Bounds
+from estimagic.optimization.tranquilo.volume import (
+    get_radius_of_cube_with_volume_of_sphere,
+)
+
+
+@dataclass
+class Region:
+    """Trust region."""
+
+    center: np.ndarray
+    radius: float
+    bounds: Bounds = None
+
+    @property
+    def shape(self) -> str:
+        any_bounds_binding = _any_bounds_binding(
+            bounds=self.bounds, center=self.center, radius=self.radius
+        )
+        return "cube" if any_bounds_binding else "sphere"
+
+    @property
+    def cube_bounds(self) -> Bounds:
+        if self.shape == "sphere":
+            radius = self.radius
+        else:
+            radius = get_radius_of_cube_with_volume_of_sphere(
+                self.radius, len(self.center), scaling_factor=1.0
+            )
+        cube_bounds = _get_cube_bounds(
+            center=self.center, radius=radius, bounds=self.bounds
+        )
+        return cube_bounds
+
+    @property
+    def cube_center(self) -> np.ndarray:
+        if self.shape == "sphere":
+            _center = self.center
+        else:
+            cube_bounds = self.cube_bounds
+            _center = (cube_bounds.lower + cube_bounds.upper) / 2
+        return _center
+
+    def map_to_unit(self, x: np.ndarray) -> np.ndarray:
+        """Map points inside the trustregion to the unit sphere or cube."""
+        return _map_to_unit(self.cube_bounds, x=x)
```

Comment:
If we define cube bounds as suggested above, we will need the if condition here. If `shape == "sphere"` -> scale by radius, if `shape == "cube"` -> scale by cube bounds.